### PR TITLE
Add support for Go-based tools

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request adds to this Ansible role support for Go-based tools.

## 💭 Motivation and context ##

Part of the resolution (along with cisagov/kali-packer#86) of cisagov/cool-system-internal#69.

## 🧪 Testing ##

All `pre-commit` hooks and `molecule` tests pass.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: tasks/install_go.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Install Go toolchain and Git
+  ansible.builtin.package:
+    name:
+      # Third-party Go packages are often installed via git
```
Comment (dav3r) [inline tasks/install_go.yml:5]: Helpful comment! 👍

**Review by dav3r — APPROVED** [review]
Strong work!  This is turning into a pretty robust Ansible role - very nice! 💎

File: tasks/install_go.yml (hunk 1 above)
Comment (jsf9k) [reply tasks/install_go.yml:5]: :bowing_man:

File: meta/main.yml [hunk 2]
```diff
@@ -6,6 +6,8 @@ galaxy_info:
   galaxy_tags:
     - assessment
     - archive
+    - go
+    - csharp
```
Comment (mcdonnnj) [inline meta/main.yml]: Just a thought.

```suggestion
    - archive
    - assessment
    - csharp
    - go
```

File: molecule/go/tests/test_go.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    assert directory.exists
+    assert directory.is_directory
+    # Make sure that the directory is not empty
+    assert host.run_expect([0], f'[ -n "$(ls -A {d})" ]')
+
+
+@pytest.mark.parametrize(
+    "pkg",
+    [
+        "golang",
```
Comment (mcdonnnj) [inline molecule/go/tests/test_go.py:34]: Just a thought for completeness with the task.

```suggestion
        "git",
        "golang",
```

**Review by mcdonnnj — APPROVED** [review]
Thanks for getting this functionality implemented. I had two suggestions for your consideration, but they don't hold back my approval. Strong work 💪

**Pull request merged**